File: C++/Model/VideoFile.cpp
#include "VideoFile.h"

// CONSTRUCTORS
VideoFile::VideoFile(string tit, string gen, string nat, int year) : title(tit), genre(gen), nation(nat), publishingYear(year) {
	// CONTROLLI DI CONSISTENZA VALORI QUI !!!
}



//  GETTERS:

string VideoFile::getTitle() const {
	return title;
}

string VideoFile::getGenre() const {
	return genre;
}

string VideoFile::getNation() const {
	return nation;
}

int VideoFile::getPublishingYear() const {
	return publishingYear;
}

//  SETTERS:

void VideoFile::setTitle(string tit) {
	// CONTROLLO SULLA LUNGHEZZA STRINGA QUI !!!
	title = tit;
}

void VideoFile::setGenre(string gen) {
	// CONTROLLO LUNGHEZZA STRINGA QUI !!!
	genre = gen;
}

void VideoFile::setPublishingYear(int year) {
	publishingYear = year;
}

void VideoFile::setNation(string nat) {
	// CONTROLLO SULLA LUNGHEZZA STRINGA QUI !!!
	nation = nat;
}

File: Java/VideoFile.java
public class VideoFile {
	private String title;
	private String genre;
	private String nation;
	private Date publishingDate;
	
	//CONSTRUCTORS:
	VideoFile(String t, String g, String n, int d, int m, int y) {
		this.title = t;
		this.genre = g;
		this.nation = n;
		this.publishingDate = new Date(d, m, y);
	}
	
	
	
	public static String getTitle() {
		return title;
	}
}
